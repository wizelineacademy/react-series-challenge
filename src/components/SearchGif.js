import React from 'react';

const SearchGif = () => {

    const onKeyUpSearch = (e) => {
        console.log('KEY_UP', e.target.value)
    };

    return (
        <div>
            <label htmlFor="searchGif">Search Gif</label>
            <input type="text" id="searchGif" onKeyUp={onKeyUpSearch}/>
        </div>
    );
};

export default SearchGif;