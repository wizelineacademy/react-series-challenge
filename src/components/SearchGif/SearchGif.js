import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import searchGifsActions from '../../actions/searchGifs';
import { SearchLabel, SearchInput  } from './Searchgif.styled';

const SearchGif = (props) => {

    const onKeyUpSearch = (e) => {
        props.loadSearchGifs(e.target.value);
    };

    return (
        <div>
            <SearchLabel htmlFor="searchGif">Search <i className="fas fa-search" /></SearchLabel>
            <SearchInput type="text" id="searchGif" onKeyUp={onKeyUpSearch}/>
        </div>
    );
};

const mapDispatchToProps = (dispatch) => {
    const { loadSearchGifs } = searchGifsActions.creators;

    return bindActionCreators({
        loadSearchGifs
    }, dispatch)
};

export default connect(null, mapDispatchToProps)(SearchGif);