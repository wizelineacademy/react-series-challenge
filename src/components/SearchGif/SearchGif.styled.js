import styled from 'styled-components'


const SearchLabel = styled.label`
    display: block;
    color: #333;
    text-align: right;
`;
SearchLabel.displayName = 'SearchLabel';

const SearchInput = styled.input`
    -webkit-border-radius: 4px;
    -moz-border-radius: 4px;
    border-radius: 4px;
    border: 1px solid #333;
    font-size: 14px;
    height: 25px;
    padding: 4px;
    
    &:focus {
        outline: none !important;
        border-color: #E93D44;
    }
`;

SearchInput.displayName = 'SearchInput';

export {
    SearchLabel,
    SearchInput,
};