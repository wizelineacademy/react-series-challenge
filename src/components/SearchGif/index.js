import React from 'react';
import { Input } from '../../styledComponents';

const SearchGif = ({
  handleSearch
}) => {
  return (
    <Input
      type="text"
      onKeyPress={handleSearch}
    />
  );
}

export default SearchGif;
