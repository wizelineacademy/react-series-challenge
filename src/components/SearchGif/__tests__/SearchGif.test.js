import React from 'react';
import { shallow } from 'enzyme';
import SearchGif from '../../SearchGif';

describe('<SearchGif />', () => {
  it('SearchGif renders without crashing', () => {
    const wrapper = shallow(<SearchGif />);
    expect(wrapper).toMatchSnapshot();
  });
});
