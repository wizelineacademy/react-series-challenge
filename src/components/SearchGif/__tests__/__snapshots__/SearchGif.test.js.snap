// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchGif /> SearchGif renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchGif />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "onKeyPress": undefined,
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "onKeyPress": undefined,
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
